packages/web-app/src/pages/proposal.tsx

```diff
@@ -76,6 +86,55 @@ const Proposal: React.FC = () => {
     }
   }, [editor, proposal]);
 
+  const findDecoder = useCallback(
+    (action: DaoAction) => {
+      const functionParams = client?.decoding.findInterface(action.data);
+      switch (functionParams?.functionName) {
+        case 'withdraw':
+          return decodeWithdrawToAction(
+            action.data,
+            client,
+            apolloClient,
+            network
+          );
+        default:
+          return {} as ActionWithdraw;
+      }
+    },
+    [apolloClient, client, network]
+  );
+
+  useEffect(() => {
+    async function getProposalAction() {
+      if (proposal) {
+        const actionPromises = proposal.actions.map(action =>
+          findDecoder(action)
+        );
+        Promise.all(actionPromises).then(value => {
+          setDecodedActions(value);
+        });
+      }
+    }
+    getProposalAction();
```

Comment:
If you're using `then` callbacks you don't need to wrap everything in an async function :)
```suggestion
      if (proposal) {
        const actionPromises = proposal.actions.map(action =>
          findDecoder(action)
        );
        Promise.all(actionPromises).then(value => {
          setDecodedActions(value);
        });
      }
```